hobos/lib/pt_lib.h

```diff
@@ -0,0 +1,84 @@
+#ifndef PT_LIB_H
+#define PT_LIB_H
+
+#include <stdint.h>
+
+//-----
+// Page Table Descriptor Attributes
+//-----
+#define PT_PAGE 	0b11        // 4k granule
+#define PT_BLOCK	0b01        // 2M granule
+
+#define PT_UXN_POS	54
+#define PT_PXN_POS	53
+#define PT_AF_POS	10
+#define PT_SH_POS	8
+#define PT_AP_POS	6
+#define PT_NS_POS	5
+#define PT_INDEX_POS	2
+
+//AP Flag (Access permissions)
+
+#define PT_KERNEL	0b00 << PT_AP_POS     // privileged, supervisor EL1 access only
```

Comment:
all the operations in defines should be with brackets, otherwise depends on expression used the operator precedence can be different from expected